core/src/main/java/bio/terra/pearl/core/service/survey/SurveyService.java

```diff
@@ -74,6 +71,16 @@ public Survey create(Survey survey) {
             AnswerMapping savedMapping = answerMappingDao.create(answerMapping);
             savedSurvey.getAnswerMappings().add(savedMapping);
         }
+
+        // parse the survey content to get the titles and create the language texts
+        Map<String, String> parsedTitles = SurveyParseUtils.parseSurveyTitle(savedSurvey.getContent(), savedSurvey.getName());
+        List<LanguageText> texts = SurveyParseUtils.titlesToLanguageTexts(
+                SurveyParseUtils.formToLanguageTextKey(savedSurvey.getStableId(),savedSurvey.getVersion()),
+                savedSurvey.getPortalId(),
+                parsedTitles);
+        languageTextDao.bulkCreate(texts);
```

Comment:
Why not do this parsing in the frontend? It seems indirect to save the content in a different format for every CRUD operation when the frontend (presumably) can just parse the survey content directly and grab the names. But I'm probably missing something that makes that more difficult / this a lot easier.